tests/components/nest/test_api.py

```diff
@@ -0,0 +1,106 @@
+"""Tests for the Nest integration API glue library."""
+
+import time
+
+from homeassistant.components.nest.api import AsyncConfigEntryAuth
+from homeassistant.components.nest.const import (
+    OAUTH2_AUTHORIZE,
+    OAUTH2_TOKEN,
+    SDM_SCOPES,
+)
+from homeassistant.helpers import config_entry_oauth2_flow
+from homeassistant.util import dt
+
+from tests.common import MockConfigEntry
+
+DOMAIN = "nest"
+
+CLIENT_ID = "some-client-id"
+CLIENT_SECRET = "some-client-secret"
+
+FAKE_TOKEN = "some-token"
+FAKE_UPDATED_TOKEN = "some-updated-token"
+FAKE_REFRESH_TOKEN = "some-refresh-token"
+
+EXPIRATION_TIME = time.time() + 86400
+
+CONFIG_ENTRY_DATA = {
+    "sdm": {},  # Indicates new SDM API, not legacy API
+    "auth_implementation": "local",
+    "token": {
+        "access_token": FAKE_TOKEN,
+        "expires_in": 12354,  # ignored
+        "refresh_token": FAKE_REFRESH_TOKEN,
+        "scope": " ".join(SDM_SCOPES),
+        "token_type": "Bearer",
+        "expires_at": EXPIRATION_TIME,
+    },
+}
+
+
+def create_auth(hass, aioclient_mock, config_entry_data):
+    """Create an AsyncConfigEntryAuth for tests."""
+    config_entry = MockConfigEntry(domain=DOMAIN, data=CONFIG_ENTRY_DATA)
+    config_entry.add_to_hass(hass)
+    implementation = config_entry_oauth2_flow.LocalOAuth2Implementation(
+        hass, DOMAIN, CLIENT_ID, CLIENT_SECRET, OAUTH2_AUTHORIZE, OAUTH2_TOKEN
+    )
+    session = config_entry_oauth2_flow.OAuth2Session(hass, config_entry, implementation)
+    return AsyncConfigEntryAuth(aioclient_mock, session, CLIENT_ID, CLIENT_SECRET)
```

Comment:
We don't want to interact with integration details in tests. Just use the core interfaces to start the tests and patch in the periphery as needed.

Reply:
We should not use `AsyncConfigEntryAuth` directly. We should just set up the integration and assert its interaction with the library by patching appropriately in the library.

Reply:
Thanks, I interpreted your comment in the opposite direction.  This has been updated to exercise everything through setting up the integration.

(I can see how maintenance on common/core components would be much easier if we don't have to worry about updating tests with extra dependencies)